src/shared/dll_load_notify.c

```diff
@@ -0,0 +1,149 @@
+/*
+ * Dll load notification helper.
+ * Copyright (C) 2015, Wazuh Inc.
+ * November 16, 2022.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#include "shared.h"
+#include "dll_load_notify.h"
+
+#ifdef WIN32
+#include <versionhelpers.h>
+
+_LdrRegisterDllNotification LdrRegisterDllNotification = NULL;
+_LdrUnregisterDllNotification LdrUnregisterDllNotifcation = NULL;
+PVOID cookie_dll_notification = NULL;
+
+/**
+ *@brief Verify all the DLLs that are already loaded
+ */
+static void loaded_modules_verification()
+{
+#if IMAGE_TRUST_CHECKS != 0
+    HMODULE handle_module[OS_SIZE_1024];
+    HANDLE handle_process = GetCurrentProcess();
+    DWORD handle_bytes_needed = 0;
+
+    if (EnumProcessModules(handle_process,
+                           handle_module,
+                           sizeof(handle_module), &handle_bytes_needed)) {
+
+        for (size_t i = 0; i < (handle_bytes_needed / sizeof(HMODULE)); i++) {
+            wchar_t module_name[MAX_PATH];
+
+            // Get the full path to the module's file.
+            if (GetModuleFileNameExW(handle_process,
+                                     handle_module[i],
+                                     module_name,
+                                     sizeof(module_name) / sizeof(wchar_t))) {
+                // Check if the images loaded are signed.
+                if (verify_hash_and_pe_signature(module_name) != OS_SUCCESS) {
+#if IMAGE_TRUST_CHECKS == 2
+                    merror_exit("The file '%S' is not signed or its signature is invalid.", module_name);
+#else
+                    mwarn("The file '%S' is not signed or its signature is invalid.", module_name);
+#endif // IMAGE_TRUST_CHECKS == 2
+                } else {
+                    mdebug1("The file '%S' is signed and its signature is valid.", module_name);
```

Comment:
Design nitpick: The format string is the same for every message. It only changes the log level.